src/Sylius/Bundle/CoreBundle/Checkout/Step/FinalizeStep.php

```diff
@@ -66,6 +68,7 @@ protected function completeOrder(OrderInterface $order)
         $this->dispatchCheckoutEvent(SyliusOrderEvents::PRE_CREATE, $order);
         $this->dispatchCheckoutEvent(SyliusCheckoutEvents::FINALIZE_PRE_COMPLETE, $order);
 
+        $this->applyTransition('complete', $order);
         $this->get('sm.factory')->get($order, OrderTransitions::GRAPH)->apply(OrderTransitions::SYLIUS_CREATE, true);
```

Comment:
We want to maintain most of current behavior to avoid too many BC breaks in a single version and too many changes in a single PR.